### PR TITLE
sdbus: make debug logging configurable

### PR DESCRIPTION
**Pull request by garlick**
Problem: sdbus is prolific in its output to the flux circular log buffer.
    
Disable debug logging by default, and allow it to be enabled by config:
```toml
[systemd]
sdbus-debug = true
````

(This was peeled off of #5197)

### CODE REVIEW

File: src/modules/sdbus/sdbus.c [hunk 1]
```diff
@@ -696,18 +759,22 @@ struct sdbus_ctx *sdbus_ctx_create (flux_t *h, flux_error_t *error)
 {
     struct sdbus_ctx *ctx;
 
-    if (!(ctx = calloc (1, sizeof (*ctx)))
-        || !(ctx->f_conn = sdbus_connect (h, true, retry_min, retry_max))
+    if (!(ctx = calloc (1, sizeof (*ctx))))
+        goto error_create;
+    if (sdbus_configure (ctx, flux_get_conf (h), error) < 0)
+        goto error;
+    if (!(ctx->f_conn = sdbus_connect (h, true, retry_min, retry_max))
```
Comment (grondo) [inline src/modules/sdbus/sdbus.c:766]: I'm probably having trouble inspecting the diff, but is the error message in `error.text` never printed if `sdbus_configure()` fails in `sdbus_ctx_create()`?

**Review by grondo — APPROVED** [review]
This LGTM, just one question inline about an error message during initialization (and I could be misreading due to the diff)

File: src/modules/sdbus/sdbus.c [hunk 2]
```diff
@@ -56,6 +56,20 @@ static void sdbus_recover (struct sdbus_ctx *ctx, const char *reason);
 static const double retry_min = 2;
 static const double retry_max = 60;
 
+static bool sdbus_debug;
```
Comment (chu11) [inline src/modules/sdbus/sdbus.c:59]: initialize to false just to be clear on default?

File: t/t2407-sdbus.t [hunk 3]
```diff
@@ -111,9 +111,22 @@ bus_subscribe_cancel () {
     flux python -c "import flux; flux.Flux().rpc(\"sdbus.subscribe-cancel\",{\"matchtag\":$1},flags=flux.constants.FLUX_RPC_NORESPONSE)"
 }
 
+test_expect_success 'enable sdbus-debug in configuration' '
```
Comment (chu11) [inline t/t2407-sdbus.t:114]: commit message "disable" -> "disabled by default"

File: src/modules/sdbus/sdbus.c (hunk 1 above)
Comment (garlick) [reply src/modules/sdbus/sdbus.c:766]: It is printed by the caller which is in main.c.  `error` is an parameter of `sdbus_ctx_create()`.

Comment (grondo) [reply src/modules/sdbus/sdbus.c:766]: Ok, thanks. sorry I missed that.

**Comment by garlick** [comment]
OK fixed those things and forced a push.  Thanks!

**Review by chu11 — APPROVED** [review]
LGTM too

**Pull request merged**